src/treeffuser/scaler.py

```diff
@@ -3,65 +3,70 @@
 
 from jaxtyping import Float
 from numpy import ndarray
-from sklearn.preprocessing import MinMaxScaler
 from sklearn.preprocessing import StandardScaler
 
 
-class Preprocessor:
+class ScalerMixedTypes:
     """
-    General lightweight preprocessor for data.
+    Data scaler for mixed data-types with continuous and categorical features.
 
-    In addition to handling simple stuff like normalization and standardization
-    of continuous data we do some lightweight standardization of categorical data.
+    - Standardize continuous data to have mean 0 and standard deviation 1 (or between -1
+      and 1 if specified).
+    - Leave categorical data unchanged if specified.
 
+    This class does not check the input data. The indices of the categorical features must be
+    provided, or they will be treated as continuous and standardized.
+
+    Parameters
+    ----------
+    scaler : scaler from sklearn.preprocessing, optional
+        The scaler to use for continuous features. Default is `StandardScaler` if not provided.
     """
 
     def __init__(
         self,
-        use_min_max: bool = False,
+        scaler=None,
     ) -> None:
-        self._scaler = None
         self._cat_idx = None
         self._is_fitted = False
         self._x_dim = None
-        self._use_min_max = use_min_max
+        if scaler is None:
+            self._scaler = StandardScaler()
 
     def fit(self, X: Float[ndarray, "batch x_dim"], cat_idx: Optional[List[int]] = None):
         """
-        Standardizes the data to have mean 0 and standard deviation 1.
-
-        This class does no checking of the input data. It is assumed that the user
-        has already checked that the input data is valid.
-
-        Args:
-        X: The data to fit the preprocessor to.
-        cat_idx: The indices of the categorical features.
+        Fit the scaler provided at initialization to the data.
+
+        Parameters
+        ----------
+        X : ndarray of shape (batch, x_dim)
+            The data to fit the preprocessor to.
+        cat_idx : list of int, optional
+            The indices of the categorical features.
         """
         self._reset()
         cat_idx = cat_idx if cat_idx is not None else []
+        non_cat_idx = X[:, [i for i in range(X.shape[1]) if i not in cat_idx]]
```

Comment:
Naming here is a bit confusing. It gives the impression that `cat_idx` and `non_cat_idx` should be complements of one another but they are really not.

Reply:
Good catch, thanks!